service/src/main/java/org/ehrbase/aql/compiler/AqlExpressionWithParameters.java

```diff
@@ -50,17 +50,20 @@ public AqlExpressionWithParameters parse(String query, Map<String, Object> param
     public AqlExpressionWithParameters parse(String expression, String jsonParameterMap) {
 
         // get the map from the json expression
-        Gson gson = new GsonBuilder().create();
+        Map<String, Object> parameterMap;
+        try {
+            parameterMap = new ObjectMapper().readerForMapOf(Object.class).readValue(jsonParameterMap);
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        }
 
-        Map<String, Object> parameterMap = gson.fromJson(jsonParameterMap, Map.class);
+        Map<String, Object> parameters = (Map<String, Object>) parameterMap.get(PARAMETERS_KEY);
 
         // get the map from query-parameters
-        if (!parameterMap.containsKey(PARAMETERS_KEY))
+        if (parameters == null) {
```

Comment:
Maybe it's good to check when there is no parames inside 

if (parameters == null || parameters.size() == 0) {

Reply:
It is OK if i's empty, but actually it looks like the code is not used at all...